### PR TITLE
[stable/grafana] fix grafana default config options

### LINKED ISSUE #1461: [stable/grafana] Default config options not applied in templates
In the grafana chart, two entries in values.yaml, `server.setDatasource.image` and `server.setDatasource.restartPolicy` are documented with defaults, but these defaults are not actually applied in the templates causing confusing errors like:

    Error: failed to create resource: Job.batch "grafana-grafana-set-datasource" is invalid: [spec.template.spec.containers[0].image: Required value, spec.template.spec.restartPolicy: Unsupported value: "Always": supported values: OnFailure, Never]

### PR DESCRIPTION
**Pull request by jwhitcraft**
Set the defaults as outlined by the values.yaml file.

fixes #1461

### CODE REVIEW

File: stable/grafana/templates/job.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
     spec:
       containers:
       - name: {{ template "grafana.server.fullname" . }}-set-datasource
-        image: "{{ .Values.server.setDatasource.image }}"
+        image: "{{ default "appropriate/curl:latest" .Values.server.setDatasource.image }}"
```
Comment (sameersbn) [inline stable/grafana/templates/job.yaml]: We recommend having the default values in the values.yaml. Please revert this change and uncomment the default value in the values.yaml file.

File: stable/grafana/templates/job.yaml [hunk 2]
```diff
@@ -40,7 +40,7 @@ spec:
           - "Content-Type: application/json;charset=UTF-8"
           - "--data-binary"
           {{- with .Values.server.setDatasource.datasource }}
-          - "{\"name\":\"{{ .name }}\",\"type\":\"{{ .type }}\",\"url\":\"{{ .url }}\",\"database\":\"{{ .database }}\",\"jsonData\":{ {{ .jsonData }} },\"access\":\"{{ .access }}\",\"isDefault\":{{ .isDefault }}}"
+          - "{\"name\":\"{{ default "default" .name }}\",\"type\":\"{{ default "prometheus" .type }}\",\"url\":\"{{ .url }}\",\"database\":\"{{ .database }}\",\"jsonData\":{ {{ .jsonData }} },\"access\":\"{{ default "proxy" .access }}\",\"isDefault\":{{ default "true" .isDefault }}}"
```
Comment (sameersbn) [inline stable/grafana/templates/job.yaml]: same as above

File: stable/grafana/templates/job.yaml [hunk 3]
```diff
@@ -40,7 +40,7 @@ spec:
...
           {{- end }}
-      restartPolicy: {{ .Values.server.setDatasource.restartPolicy }}
+      restartPolicy: {{ default "OnFailure" .Values.server.setDatasource.restartPolicy }}
```
Comment (sameersbn) [inline stable/grafana/templates/job.yaml]: same as above

**Review by sameersbn — CHANGES_REQUESTED** [review]

**Comment by jwhitcraft** [comment]
@sameersbn I updated this pr per your comments.

**Review by sameersbn — APPROVED** [review]

**Pull request merged**